pyproject.toml

```diff
@@ -29,6 +31,16 @@ select = [
 ignore = [
     # incompatible with sqlalchemy https://github.com/astral-sh/ruff/issues/4560
     "E712",
+    # code complexity checks that we fail
+    "PLR0912", "PLR0913", "PLR0915",
```

Comment:
(Would there be any benefit to configuring them with a different max in the pyproject.toml instead of completely ignoring them? i.e. nudge us towards better future behaviour?)

Reply:
I'm generally just bearish on automated code complexity checks (e.g. I trust that if you're creating a function that takes 6+ arguments, you have a good reason to do so) which is why I opted to disable entirely rather than raise the limits, but I'm not against that if other people find them useful.

Reply:
IMO those checks are most useful when wrangling different kinds of outside contributors and trying to nudge folks in a certain direction prior to giving manual review feedback, but I'm fine to leave it as is.